satpy/readers/geos_netcdficare.py

```diff
@@ -0,0 +1,769 @@
+# import math
+
+
+class NETCDF_ICARE(BaseFileHandler) :
+	# Cf readers/file_handlers.py.
+
+	def __init__(self, filename, filename_info, filetype_info) :
+		"""Init the file handler."""
+
+		super().__init__(filename, filename_info, filetype_info)
+
+		# chunk_bytes = self._chunk_bytes_for_resolution()
+		chunk_bytes = '128 MiB'
+		# chunk_bytes = '64 MiB'
+
+		# with dask.config.set(**{'array.slicing.split_large_chunks': True}) :
+		with dask.config.set({"array.chunk-size": chunk_bytes}) :
+			f_obj = open_file_or_filename(self.filename)
+			self.nc = xr.open_dataset(
+				f_obj, decode_cf=True, mask_and_scale=False,
+				chunks={"xc": "auto", "yc": "auto"})
+
+		self.metadata = {}
+
+		self.metadata["start_time"] = self.get_endOrStartTime(
+			"time_coverage_start")
+		self.metadata["end_time"] = self.get_endOrStartTime(
+			"time_coverage_end")
+
+		print(
+			"Reading: {}".format(filename),
+			" start: {} ".format(self.start_time),
+			" end: {}".format(self.end_time))
+		self._cache = {}
+
+		self.sensor_name()
+		self.res()
+		self.longitudeReelle = self.satlon()
+		self.longitudedeprojection = self.projlon()
+
+		self.zone = self.nc.attrs["Area_of_acquisition"]
+		# globe, europe.
+
+		# Reading the needed attributes.
+		self.initialisation_dataset()
+		# __init__()
+
+	def sensor_name(self) :
+		"""Get the sensor name.
+		The sensor and platform names are stored together, eg: MSG1/SEVIRI
+		"""
+		variable = self.nc["satellite"]
+		self.plateforme = variable.attrs["id"]  # msg1, msg01, MSG1...
+		platform = self.plateforme[:3]		# msg, MSG
+		platform = platform.lower()		# msg
+
+		pdict = {}
+		pdict["msg"] = "seviri"
+		pdict["mtg"] = "fci"
+		pdict["goe"] = "abi"
+		pdict["him"] = "ahi"
+
+		if platform in pdict :
+			self.sensor = pdict[platform]
+		else :
+			print(
+				"Unsupported satellite platform : " +
+				self.plateforme)
+			exit(1)
+
+		# Icare and météo france use non-standard platform names.
+		# Change is needed for pyspectral :
+		# pyspectral/rsr_seviri_Meteosat-10.h5 in the call
+		# Calculator(platform_name, sensor, name).
+		pdict = {}
+		pdict["msg1"] = "Meteosat-08"
+		pdict["msg01"] = "Meteosat-08"
+		pdict["MSG1"] = "Meteosat-08"
+		pdict["msg2"] = "Meteosat-09"
+		pdict["msg02"] = "Meteosat-09"
+		pdict["MSG2"] = "Meteosat-09"
+		pdict["msg3"] = "Meteosat-10"
+		pdict["msg03"] = "Meteosat-10"
+		pdict["MSG3"] = "Meteosat-10"
+		pdict["msg4"] = "Meteosat-11"
+		pdict["msg04"] = "Meteosat-11"
+		pdict["MSG4"] = "Meteosat-11"
+		pdict["mtgi1"] = "Meteosat-12"
+		pdict["mtg1"] = "Meteosat-12"
+		pdict["MTG01"] = "Meteosat-12"
+		pdict["goes16"] = "GOES-16"
+		pdict["goes17"] = "GOES-17"
+		pdict["goes18"] = "GOES-18"
+		pdict["goes19"] = "GOES-19"
+		pdict["hima08"] = "Himawari-8"
+		pdict["hima09"] = "Himawari-9"
+
+		if self.plateforme in pdict :
+			self.plateforme = pdict[self.plateforme]
+		else :
+			print(
+				"Unsupported satellite platform : " +
+				self.plateforme)
+			exit(1)
+
+		# sensor_name()
```

Comment:
This function should return the sensor name. Also, we try to avoid setting attributes in methods other than `__init__`